### PR TITLE
Make thrift retry configurable

### PR DESCRIPTION
**Pull request by hellium01**

### CODE REVIEW

File: presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftConnectorConfig.java [hunk 1]
```diff
@@ -83,4 +93,74 @@ public ThriftConnectorConfig setLookupRequestsConcurrency(int lookupRequestsConc
         this.lookupRequestsConcurrency = lookupRequestsConcurrency;
         return this;
     }
+
+    @Config("presto-thrift.max-retry-attempts")
+    public ThriftConnectorConfig setMaxRetryAttempts(int maxRetryAttempts)
```
Comment (AlekseiS) [inline presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftConnectorConfig.java]: The convention is "get" method and then "set" method for the same parameter.

File: presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java [hunk 2]
```diff
@@ -50,19 +49,19 @@
     private final RetryDriver retry;
 
     @Inject
-    public RetryingPrestoThriftServiceProvider(@NonRetrying PrestoThriftServiceProvider original, @ForRetryDriver ListeningScheduledExecutorService retryExecutor)
+    public RetryingPrestoThriftServiceProvider(ThriftConnectorConfig config, @NonRetrying PrestoThriftServiceProvider original, @ForRetryDriver ListeningScheduledExecutorService retryExecutor)
```
Comment (AlekseiS) [inline presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java]: Typically you add a list of parameters to the end

File: presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java [hunk 3]
```diff
@@ -50,19 +49,19 @@
...
     {
         this.original = requireNonNull(original, "original is null");
         requireNonNull(retryExecutor, "retryExecutor is null");
 
         retry = RetryDriver.retry(retryExecutor)
-                .maxAttempts(5)
+                .maxAttempts(config.getMaxRetryAttempts())
```
Comment (AlekseiS) [inline presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java]: before it you need to do `requireNonNull(config, ...)`

**Comment by hellium01** [comment]
Updated the PR to address the comments, @AlekseiS , please let me know what else should be changed.

**Pull request closed without merging**